app/src/main/java/com/stevesoltys/seedvault/settings/SettingsManager.kt

```diff
@@ -132,6 +144,15 @@ class SettingsManager(private val context: Context) {
 
     fun isBackupEnabled(packageName: String) = !blacklistedApps.contains(packageName)
 
+    fun isAppAllowedForBackup(packageName: String): Boolean {
+        // Check that the app is not blacklisted by the user
+        val enabled = isBackupEnabled(packageName)
+        if (!enabled) Log.w(TAG, "Backup of $packageName disabled by user.")
+        // We need to exclude the DocumentsProvider used to store backup data.
+        // Otherwise, it gets killed when we back it up, terminating our backup.
+        return enabled && packageName != pluginProviderPackageName
```

Comment:
Nitpick: If you add `&& packageName != pluginProviderPackageName` part above, then the `pluginProviderPackageName` gets logged as disabled as well (would need to change log message). Maybe doesn't need to be logged as warning as it is expected behavior?